src/Microsoft.ML.DataView/KeyDataViewType.cs

```diff
@@ -9,31 +9,64 @@
 namespace Microsoft.ML.Data
 {
     /// <summary>
-    /// KeyDataViewTypes are for "id"-like data. The information happens to be stored in an unsigned integer
-    /// type, but the information is not inherently numeric, so, typically, arithmetic is not
-    /// meaningful. Examples are SSNs, phone numbers, auto-generated/incremented key values,
-    /// class numbers, etc. For example, in multi-class classification, the label is typically
-    /// a class number which is naturally a KeyDataViewType.
+    /// This type is for data representing some enumerated value. This is an enumeration over a defined, known
+    /// cardinality set, as expressed through <see cref="Count"/>. The underlying .NET type is one of the unsigned
+    /// integer types. Most commonly this will be <see cref="uint"/>, but could alternately be <see cref="byte"/>,
+    /// <see cref="ushort"/>, or <see cref="ulong"/>. Despite this, the information is not inherently numeric, so,
+    /// typically, arithmetic is not meaningful. For example, in multi-class classification, the label is typically a
+    /// class number which is naturally a <see cref="KeyDataViewType"/>.
     ///
-    /// KeyDataViewTypes have a cardinality (i.e., <see cref="Count"/>) that is strictly positive.
+    /// Note that for data of this type, a value of 0, being the default value of the representation type, indicates
+    /// the missing value since it would not be sensible for the default value to correspond to any one particular specific
+    /// value of the set. The first non-missing value for the enumeration of the set is always <c>1</c>.
     ///
-    /// Note that the underlying representation value does not necessarily match the logical value.
-    /// For example, if a KeyDataViewType has range 0-5000, then it has a <see cref="Count"/> of 5001, but
-    /// the representational values are 1-5001. The representation value zero is reserved
-    /// to mean a missing value (similar to NaN).
+    /// For instance, if you had a key value with a <see cref="Count"/> of 3, then the <see cref="uint"/> value <c>0</c>
+    /// would correspond to the missing key value, and one of the values of <c>1</c>, <c>2</c>, or <c>3</c> would be one
+    /// of the valid values, and no other values should in principle be used.
+    ///
+    /// Note that in usage and structure, this is quite close in intended usage and structure to so-called "factor
+    /// variables" in R.
     /// </summary>
     public sealed class KeyDataViewType : PrimitiveDataViewType
     {
+        /// <summary>
+        /// Constructor for this type.
```

Comment:
(nit) usually these are worded as:

```
Initializes a new instance of the XXX class
```

Reply:
Yeah, I always hated those. "Oh I thought this was a constructor for some other type, gee whiz I'm glad that was clarified." But, you know, I'll change it. :)